src/main/java/org/jnode/fs/xfs/XfsDirectory.java

```diff
@@ -96,10 +96,10 @@ protected FSEntryTable readEntries() throws IOException {
                     entryCount += entries.size();
 
                     while (entries.size() < entryCount) {
-                        ShortFormDirectoryEntry entry = new ShortFormDirectoryEntry(inode.getData(), offset, recordSize);
+                        ShortFormDirectoryEntry entry = new ShortFormDirectoryEntry(inode.getData(), offset, recordSize, fileSystem);
```

Comment:
I don't think it's necessary to pass in the filesystem just to determine version. Maybe just a boolean?